saleor/discount/utils/order.py

```diff
@@ -48,19 +53,56 @@ def create_or_update_discount_objects_from_promotion_for_order(
     lines_info: list["EditableOrderLineInfo"],
     database_connection_name: str = settings.DATABASE_CONNECTION_DEFAULT_NAME,
 ):
-    create_order_line_discount_objects_for_catalogue_promotions(lines_info)
-    # base unit price must reflect all actual catalogue discounts
+    delete_order_line_discount_objects_for_catalogue_promotions(lines_info)
+    # Update base price with catalogue discount reward. This price will be further used:
+    # - to qualify order for order promotion
+    # - to apply vouchers
     _update_base_unit_price_amount_for_catalogue_promotion(lines_info)
```

Comment:
Do we still need this line? It makes:
```python
line_info.line.base_unit_price = line_info.variant_discounted_price
```
But as we operating on denormalized values, this looks like something not needed 🤔

Reply:
This ensure the proper order of discount application. First we apply catalogue discounts what results in first decreasing of line base price. Next we apply vouchers or order promotion to the decreased base prices.

The next two PRs will handle vouchers and order promotions, so maybe it will be removed. Do you suggest to handle all the disocunts in single PR?

Reply:
I am just thinking about the current logic called inside `fetch_order_prices_if_expired`. As we agreed that we should use de-normalized values on Order/OrderLine models, it doesn't make sense to make any calculations that would change the `base_price`, or any other fields that store the Money representation.
For example, what is the reason for calling `create_or_update_discount_objects_for_order` inside `fetch_order_prices_if_expired` in case of making an update on `shippingAddress`?

Reply:
I don't know what is a rationale of invalidating prices after `shippingAddress` update, but let's have an example of updating `shippingMethod`. This for sure results in different shipping price. Let's look on examples:
1. We have shipping voucher applied to order -> we need to do an update of shipping discount amount. Where would you like to do the update?
2. Increased shipping price results in higher total price of an order. This change can qualify the order for some order promotion.

Next PR will handle vouchers to use denormalised discount prices. It will require to move the whole logic of voucher application out of  `fetch_order_prices_if_expired`. This change will probably get rid of manipulating with `base_price` inside the `create_or_update_discount_objects_for_order`.